controllers/movies.js

```diff
@@ -0,0 +1,50 @@
+const movies = require('../movies')
+
+const getAllMovies = (request, response) => {
+  return response.send(movies)
+}
+
+const getMoviesByDirectors = ('/:director', (request, response) => {
+  const { director } = request.params
+
+  const allMoviesByDirectors = movies.filter((movies) => {
+    return movies.director.toLowerCase().includes(director.toLowerCase())
+  })
```

Comment:
You redefine the term `movies`. `movies` is your array of movies provided by the data file. You then call filter on it, filter will call your inner function for every movie in your list passing it a single movie at a time (15 movies in array, 15 calls to inner function). You named the single movie object being passed by filter `movies` as well and that causes scope issues.